### PR TITLE
logictest: de-flake distsql_enum

### LINKED ISSUE #93895: pkg/sql/logictest/tests/5node-disk/5node-disk_test: TestLogic_distsql_enum failed
pkg/sql/logictest/tests/5node-disk/5node-disk_test.TestLogic_distsql_enum [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8017611?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8017611?buildTab=artifacts#/) on master @ [b6667d9c6ea091303cf02ed2050e0350eada5bcb](https://github.com/cockroachdb/cockroach/commits/b6667d9c6ea091303cf02ed2050e0350eada5bcb):

```
=== RUN   TestLogic_distsql_enum
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/6e9d563e9f7703abdf07517fb35efb2e/logTestLogic_distsql_enum782437834
    test_log_scope.go:79: use -show-logs to present logs inline
[10:28:40] setting distsql_workmem='27818B';
    logic.go:2860: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1345/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node-disk/5node-disk_test_/5node-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum:68: EXPLAIN (VEC)
        SELECT x from t1 WHERE EXISTS (SELECT x FROM t2 WHERE t1.x=t2.x AND t2.y='hello')
        expected:
            │
            ├ Node 1
            │ └ *colrpc.Outbox
            │   └ *rowexec.joinReader
            │     └ *colfetcher.ColBatchScan
            ├ Node 2
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   ├ *rowexec.joinReader
            │   │ └ *colfetcher.ColBatchScan
            │   └ *colrpc.Inbox
            └ Node 3
              └ *colrpc.Outbox
                └ *rowexec.joinReader
                  └ *colfetcher.ColBatchScan
            
        but found (query options: "nodeidx=1,retry") :
            │
            ├ Node 1
            │ └ *colrpc.Outbox
            │   └ *rowexec.joinReader
            │     └ *colfetcher.ColBatchScan
            └ Node 2
              └ *colexec.ParallelUnorderedSynchronizer
                ├ *colrpc.Inbox
                └ *rowexec.joinReader
                  └ *colfetcher.ColBatchScan
[10:29:27] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1345/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node-disk/5node-disk_test_/5node-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum: 8 statements
[10:29:27] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1345/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node-disk/5node-disk_test_/5node-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum with config 5node-disk: 8 tests, 1 failures
    logic.go:3990: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1345/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node-disk/5node-disk_test_/5node-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum:88: error while processing
    logic.go:3990: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1345/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/5node-disk/5node-disk_test_/5node-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/distsql_enum:88: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/6e9d563e9f7703abdf07517fb35efb2e/logTestLogic_distsql_enum782437834
--- FAIL: TestLogic_distsql_enum (49.03s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_distsql_enum.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-22590

### PR DESCRIPTION
**Pull request by yuzefovich**
This commit de-flakes `distsql_enum` test. Previously, in a regression test we relied on injected table stats to get to the desired plan with a lookup join, but this commit switches to using the join hint to get there.

Fixes: #93895.

Release note: None

### CODE REVIEW

**Comment by yuzefovich** [comment]
I've stressed this for a bit, and no flake occurred:
```
3216 runs so far, 0 failures, over 14m10s
```
However, we couldn't repro the original flake either, so this fix is somewhat speculative.

File: pkg/sql/logictest/testdata/logic_test/distsql_enum [hunk 1]
```diff
@@ -44,48 +44,46 @@ ALTER TABLE t2 SPLIT AT VALUES(0),(10),(20);
...
-ALTER TABLE t1 INJECT STATISTICS '[
-  {
-    "columns": ["x"],
-    "created_at": "2018-01-01 1:00:00.00000+00:00",
-    "row_count": 10,
-    "distinct_count": 100
-  }
-]'
+query TTTI colnames
+SELECT start_key, end_key, replicas, lease_holder from [SHOW RANGES FROM TABLE t1 WITH DETAILS] ORDER BY lease_holder
```
Comment (mgartner) [inline pkg/sql/logictest/testdata/logic_test/distsql_enum:48]: Why did you add these new tests?

**Review by mgartner — APPROVED** [review]
LGTM

**Review by yuzefovich — COMMENTED** [review]
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94350#-:-NKLU3LT597aBmOHJVc4:b-oduykh)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @mgartner)

___
*[`pkg/sql/logictest/testdata/logic_test/distsql_enum` line 48 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/94350#-NKLCyJTkbXXyBrt1tZM-r1-48:-NKLTfUr-9tHE2gxD9Xx:bx3i42i) ([raw file](https://github.com/cockroachdb/cockroach/blob/69133d9852dd9f40336f7e7cb424f473fa844303/pkg/sql/logictest/testdata/logic_test/distsql_enum#L48)):*
<details><summary><i>Previously, mgartner (Marcus Gartner) wrote…</i></summary><blockquote>

Why did you add these new tests?
</blockquote></details>

I think it's a good rule of thumb whenever we split and relocate ranges manually in logic tests to verify the placement of those ranges via `SHOW RANGES`, to make sure the placement is as expected.

**Review by mgartner — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94350)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[`pkg/sql/logictest/testdata/logic_test/distsql_enum` line 48 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/94350#-NKLCyJTkbXXyBrt1tZM-r1-48:-NKLWNqNDlgWc44XuhhQ:b-9c9uac) ([raw file](https://github.com/cockroachdb/cockroach/blob/69133d9852dd9f40336f7e7cb424f473fa844303/pkg/sql/logictest/testdata/logic_test/distsql_enum#L48)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

I think it's a good rule of thumb whenever we split and relocate ranges manually in logic tests to verify the placement of those ranges via `SHOW RANGES`, to make sure the placement is as expected.
</blockquote></details>

SGTM. Thanks for explaining!

**Comment by yuzefovich** [comment]
A flake.

bors r+

**Comment by michae2** [comment]
blathers backport release-22.2
